pkg/build/controller/build/build_controller.go

```diff
@@ -881,12 +883,24 @@ func (bc *BuildController) createBuildPod(build *buildapi.Build) (*buildUpdate,
 			bc.recorder.Eventf(build, kapi.EventTypeWarning, "FailedCreate", "Pod already exists: %s/%s", buildPod.Namespace, buildPod.Name)
 			glog.V(4).Infof("Build pod %s/%s for build %s already exists", build.Namespace, buildPod.Name, buildDesc(build))
 
-			// If the existing pod was created before this build, switch to the Error state.
+			// If the existing pod was not created by this build, switch to the
+			// Error state.
 			existingPod, err := bc.podClient.Pods(build.Namespace).Get(buildPod.Name, metav1.GetOptions{})
-			if err == nil && existingPod.CreationTimestamp.Before(&build.CreationTimestamp) {
+			if err == nil && !strategy.HasOwnerReference(existingPod, build) {
 				update = transitionToPhase(buildapi.BuildPhaseError, buildapi.StatusReasonBuildPodExists, buildapi.StatusMessageBuildPodExists)
 				return update, nil
 			}
+			// QUESTION 1: isn't it a mistake to be returning nil, nil here in
+			// the case that err != nil?  Shouldn't we return nil, err in that
+			// case?
```

Comment:
I would say that yes it's a mistake to return nil, nil in case err != nil ... we want to retry handling this build.

Reply:
+1